manim/utils/hashing.py

```diff
@@ -0,0 +1,150 @@
+            return list(obj)
+        elif hasattr(obj, "__dict__"):
+            return getattr(obj, '__dict__')
+        elif isinstance(obj, np.uint8):
+            return int(obj)
+        try:
+            return json.JSONEncoder.default(self, obj)
+        except TypeError:
+            # This is used when the user enters an unknown type in CONFIG. Rather than throwing an error, we transform
+            # it into a string "Unsupported type for hashing" so it won't affect the hash.
+            return "Unsupported type for hashing"
+
+
+def get_json(object):
+    """Recursively serialize object to JSON. Use CustomEncoder class above.
+
+    Paramaters
+    ----------
+    dict_config : :class:`dict`
+        The dict to flatten
+
+    Returns
+    -------
+    `str` 
+        the object flattened
+    """
+    return json.dumps(object, cls=CustomEncoder)
+
+
+def get_camera_dict_for_hashing(camera_object):
+    """Remove some keys from cameraobject.__dict__ that are useless for the caching functionnality and very heavy. 
+
+    Parameters
+    ----------
+    object_camera : :class:``~.Camera`
+        The camera object used in the scene
+
+    Returns
+    -------
+    `dict`
+        Camera.__dict__ but cleaned.
+    """
+    camera_object_dict = copy.copy(camera_object.__dict__)
+    # We have to clean a little bit camera_dict, as pixel_array and background are two very big numpy array.
+    # They are not essential to caching process.
+    # We also have to remove pixel_array_to_cairo_context as it conntains uses memory adress (set randomly). See l.516 get_cached_cairo_context in camera.py
+    for to_clean in ['background', 'pixel_array', 'pixel_array_to_cairo_context']:
+        camera_object_dict.pop(to_clean, None)
+    return camera_object_dict
+
+
+def get_hash_from_play_call(camera_object, animations_list, current_mobjects_list):
+    """Take the list of animations and a list of mobjects and output their hash. Is meant to be used for `scene.play` function.
+
+    Parameters
+    -----------
+    object_camera : :class:``~.Camera`
+        The camera object used in the scene
+
+    animations_list : :class:`list`
+        The list of animations 
+
+    current_mobjects_list : :class:`list`
+        The list of mobjects.
+
+    Returns
+    -------
+    `str` 
+        concatenation of the hash of object_camera, animations_list and current_mobjects_list separated by '_'.
+    """
+    camera_json = get_json(get_camera_dict_for_hashing(camera_object))
+    animations_list_json = [get_json(x) for x in sorted(
+        animations_list, key=lambda obj: str(obj))]
+    current_mobjects_list_json = [get_json(x) for x in sorted(
+        current_mobjects_list, key=lambda obj: str(obj))]
+    hash_camera = zlib.crc32(repr(camera_json).encode())
+    hash_animations = zlib.crc32(repr(animations_list_json).encode())
+    hash_current_mobjects = zlib.crc32(
+        repr(current_mobjects_list_json).encode())
```

Comment:
We gotta make this code more DRY
```suggestion
    hash_camera, hash_animations, hash_current_mobjects = [
        zlib.crc32(repr(json_val).encode())
        for json_val in (camera_json, animations_list_json, current_mobjects_list_json)
    ]
```